### PR TITLE
RSLC-5 As a user I should be able to delete a previously created target

### PR DESCRIPTION
**Pull request by diegoxanderdx**
#### Board:
RSLC-5(https://trello.com/c/KtY88PMO)
---
#### Description:
As a user I should be able to delete a previously created target so it's no longer available in the map
---
#### Notes:
Please check the following commit
[add destroy action for user target](https://github.com/diegoxanderdx/Target/pull/7/commits/26141deb94886f0a009e8fdca76e2fa8b64ed362)
---
#### Tasks:
- [x] Create destroy spec
- [x] Edit policies for targets 
- [x] Edit targets controller
---
#### Risk:
*
---

### CODE REVIEW

File: app/controllers/api/v1/targets_controller.rb [hunk 1]
```diff
@@ -11,6 +11,12 @@ def create
         @target = current_user.targets.create!(target_params)
       end
 
+      def destroy
+        @target = current_user.targets.find(params[:id])
+        authorize @target
```
Comment (enzofab91) [inline app/controllers/api/v1/targets_controller.rb:16]: what about adding `return unless @target.present?` before this block?

File: spec/requests/api/v1/targets/destroy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  let(:id) { first_target.id }
+  subject { delete api_v1_target_path(id:), headers: auth_headers, as: :json }
+
+  it 'returns a successful response' do
+    subject
+    expect(response).to be_successful
+  end
+
+  it 'deletes the target' do
+    expect { subject }.to change(user.targets, :count).from(3).to(2)
```
Comment (enzofab91) [inline spec/requests/api/v1/targets/destroy_spec.rb:15]:
```suggestion
    expect { subject }.to change(user.targets, :count).by(-1)
```

**Review by enzofab91 — APPROVED** [review]
Nice!

File: spec/requests/api/v1/targets/destroy_spec.rb (hunk 2 above)
Comment (diegoxanderdx) [reply spec/requests/api/v1/targets/destroy_spec.rb:15]: cool

**Review by brunoparma88 — APPROVED** [review]

**Pull request merged**